### PR TITLE
Add support for AndroidProof v2

### PR DESCRIPTION
**Pull request by DavisJP**
Change the signature algorithm from EC to RSA.
Add different config file for v2 with Certificate chain, apk hashes and Google API key values.
Update requestId to parsed hex values.
Update web Authenticity checks from v1, it was incomplete.
Add support for multiple Google API keys for multiple accounts
Add androidProofV2 sample
Add note on Readme for AndroidProof config files

### CODE REVIEW

**Comment by Fi3** [comment]
For the commit message  try to follow https://github.com/Fi3/h/wiki/Commit

The content of `settings/android-proof-config-v2.json` should be in `settings/settings.json` if you have conflict you can change the structure of `seetings` in something like `{v1: {}, v2: {}}` or just use differents names for example instead of `googleApiKey` use `googleApiKeyV1`.

A lot of shared code between `src/android-verify.js ` and `add-support-for-androidproof_v2 ` try to be more DRY. Are you sure that we need two modules for the two versions? (I'm not saying that should be only one module, just asking)

**Comment by DavisJP** [comment]
Which parts of the commit message do you think I can improve?

I like the idea of putting the settings all in one file, I'll have a look at that.

I completely agree with you, the thing is that I did my best to avoid changing anything on how the v1 is processed because I don't know how the Proof Verification Tool is used when embedded and given my very little experience with Js I thought it would be better this way. Maybe we can refactor later once we get everything tested, but I will need some pointers on best practices of how to do these changes.

Thanks for the comments Fillippo.

File: src/android-verify.js [hunk 1]
```diff
@@ -6,23 +6,40 @@ var request = require('isomorphic-fetch');
...
+  const buf = new Buffer(cborEncodedData.buffer);
+  var androidProof = cbor.decodeFirstSync(buf);
+
+  let settings;
+  let requestID;
+  let encodedChain;
+  if (version === 'v1') {
+    settings = jsonSettings.v1;
+    requestID = androidProof.requestID.toString();
+    encodedChain = new Buffer(jsonSettings.v1.androidCertChain, "base64");
```
Comment (Fi3) [inline src/android-verify.js]: use single quote (style consistency purpose)

File: src/android-verify.js [hunk 2]
```diff
@@ -6,23 +6,40 @@ var request = require('isomorphic-fetch');
...
+  } else if (version === 'v2'){
+    settings = jsonSettings.v2;
+    requestID = androidProof.requestID.toString('hex');
+    encodedChain = new Buffer(jsonSettings.v2.androidCertChain, "base64");
```
Comment (Fi3) [inline src/android-verify.js]: use single quote (style consistency purpose)

File: src/android-verify.js [hunk 3]
```diff
@@ -52,30 +69,56 @@ export const verify = async (data: Uint8Array) => {
...
     }
   }
 
-  verifySignature(jws, googleCert);
-  await verifyAuthenticity(jws, googleApiKey);
+  if (!verifySignature(jws, googleCert)) {
+    throw new Error('verifySignature failed');
+  }
+
+  for (var i = 0; i < googleApiKey.length; i++) {
```
Comment (Fi3) [inline src/android-verify.js]: always use `let` over `var`. If you want know the reason try open the node repl and try `for(var i=0; i<10; i++){console.log(9)}` then `for(let k=0; k<10; k++){console.log(9)}` then try to print `i` and `k`

File: src/android-verify.js [hunk 4]
```diff
@@ -52,30 +69,56 @@ export const verify = async (data: Uint8Array) => {
...
+        throw new Error('version unsupported');
+      }
+      
+      if (result) {
+        status = 'valid';
+        break;
+      }
+    } catch (err) {
+      if (err.message != 200) {
+        status = err.message
```
Comment (Fi3) [inline src/android-verify.js]: you should end the line with `;` just style consistency purpose

File: src/android-verify.js [hunk 5]
```diff
@@ -52,30 +69,56 @@ export const verify = async (data: Uint8Array) => {
...
+      }
+      
+      if (i == googleApiKey.length - 1) {
+        throw new Error('verifyAuthenticity failed')
```
Comment (Fi3) [inline src/android-verify.js]: you should end the line with `;` just style consistency purpose

File: src/android-verify.js (hunk 3 above)
Comment (DavisJP) [reply src/android-verify.js]: Wow, js is really weird... thanks for pointing this out!

**Comment by DavisJP** [comment]
Added final changes.

File: src/android-verify.js [hunk 6]
```diff
@@ -6,23 +6,40 @@ var request = require('isomorphic-fetch');
 var asn = require('asn1.js');
 const jsonSettings = require('../settings/settings.json');
 const R = require('ramda');
-import {AndroidProof} from '../certs/AndroidProof.js';
 // $FlowFixMe
 const Buffer = require('buffer').Buffer;
 
 
-export const verify = async (data: Uint8Array) => {
+export const verify = async (data: Uint8Array, version: string) => {
```
Comment (Fi3) [inline src/android-verify.js]: Here is better to use a declared type over a generic one `type AndroidVersion = 'v1' | 'v2'`. Not a blocker

**Comment by Fi3** [comment]
Weel done! :+1:

File: src/android-verify.js (hunk 6 above)
Comment (DavisJP) [reply src/android-verify.js]: Good to know! I'll keep that in mind for future developments, thanks!

**Pull request merged**